--- a/lib/eigenlayer-middleware/lib/eigenlayer-contracts/src/contracts/libraries/SlashingLib.sol
+++ b/lib/eigenlayer-middleware/lib/eigenlayer-contracts/src/contracts/libraries/SlashingLib.sol
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.27;
 
-import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../../../lib/openzeppelin-contracts-v4.9.0/contracts/utils/math/Math.sol";
 import "@openzeppelin-upgrades/contracts/utils/math/SafeCastUpgradeable.sol";
 
 /// @dev All scaling factors have `1e18` as an initial/default value. This value is represented

