--- a/foundry.toml
+++ b/foundry.toml
@@ -20,7 +20,7 @@
         "@eigenlayer/=lib/eigenlayer-middleware/lib/eigenlayer-contracts/src/",
         "@eigenlayer-scripts/=lib/eigenlayer-middleware/lib/eigenlayer-contracts/script/",
         "@eigenlayer-middleware/=lib/eigenlayer-middleware/",
-        "@openzeppelin/=lib/eigenlayer-middleware/lib/openzeppelin-contracts/",
+        "@openzeppelin/=lib/eigenlayer-middleware/lib/eigenlayer-contracts/lib/openzeppelin-contracts-v4.9.0/",
         "@openzeppelin-upgrades/=lib/eigenlayer-middleware/lib/openzeppelin-contracts-upgradeable/",
         "forge-std/=lib/forge-std/src/",
         "@risc0/=lib/risc0-ethereum/contracts/src",

